### PR TITLE
feat/admin-page

### PR DESCRIPTION
**Pull request by alivadjid**
First decision of issue to [add admin page](https://github.com/DamyanBG/real-estate-react-app/issues/102)
Now i'm thinking about admin role

### CODE REVIEW

**Comment by alivadjid** [comment]
I think set user role in frontend - is not good practise. Mayb better set at the backend role - 'user' by default?

**Comment by DamyanBG** [comment]
Maybe I have to tell you more behind the business logic, then we will think for the implementation. A user is able to register as user (later we can call him customer or something like that) and seller. Additionally a user can be an admin, but admins will be created by the person, who owns the back and and the database and by users, which are already admins.

So, the logic behind the back end is - everybody is able to register like user or seller, but registration as admin will be forbidden. When an user login into the back end, he receives a token + the user info from the back end. We will store the role of the user in the local storage, because we have to separate the logic for every yser somehow. So, the role is not set from the front end, I hope this answers to your question.

If somebody is smarter and changes the role manually in the front end local storage, he will be able to see the admin panel, you are right. But, the back end will wait for token, which was created for admin, so the back end points are restricted and the user can do nothing.

**Comment by alivadjid** [comment]
> Maybe I have to tell you more behind the business logic, then we will think for the implementation. An user is able to register as user (later we can call him customer or something like that) and seller. Additionally an user can be an admin, but admins will be created by the person, who owns the back and and the database and by users, which are already admins.
> 
> So, the logic behind the back end is - everybody is able to register like user or seller, but registration as admin will be forbidden. When an user login into the back end, he receives a token + the user info from the back end. We will store the role of the user in the local storage, because we have to separate the logic for every yser somehow. So, the role is not set from the front end, I hope this answers to your question.
> 
> If somebody is smarter and changes the role manually in the front end local storage, he will be able to see the admin panel, you are right. But, the back end will wait for token, which was created for admin, so the back end points are restricted and the user can do nothing.

ok, i got it. Changed status to draft, i want to finish issue at all and show user list. Try to find route at the backend to get all users

**Comment by alivadjid** [comment]
Issue done. 
But in the end all my request gets Error by CORS

**Comment by DamyanBG** [comment]
> Issue done. 
> But in the end  something went wrong. All my requests gets Error by CORS

I will check tomorrow.

File: src/context/UserContext.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
 import { createContext, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
+import { hostUrl } from '../common/urls';
 
 export const UserContext = createContext();
 
 export const UserProvider = ({ children }) => {
     const [user, setUser] = useState(null);
+    const [userInfo, setUserInfo] = useState({});
```
Comment (DamyanBG) [inline src/context/UserContext.js:9]: I will approve that for now, but this code should not be here.

**Review by DamyanBG — APPROVED** [review]

File: src/context/UserContext.js (hunk 1 above)
Comment (alivadjid) [reply src/context/UserContext.js:9]: Where better to be? I tried to save userInfo here.

**Comment by alivadjid** [comment]
> > Issue done.
> > But in the end  something went wrong. All my requests gets Error by CORS
> 
> I will check tomorrow.

Something still happend with backend.
изображение
изображение
изображение

**Comment by DamyanBG** [comment]
Back end is not available, because last night my free account was ended. So today I paid to continue using Azure. Now I will fix everything.

**Comment by DamyanBG** [comment]
https://github.com/DamyanBG/real-estate-react-app/pull/119/files look this and learn

**Pull request merged**